Added logrus logger + passing a name to HelloWorldRequest

allaptr opened the pull request:
Small, low hanging fruit, improvements. 
- Replaced default logger with "github.com/sirupsen/logrus", set default log level to Info, configurable by passing a flag into service main
- Added instructions to the README for turning on gRPC logging, Info level, which seems to be the most verbose
- Passing the name value to the HelloWorldRequest, which then gets received and echoed by the server

Commit:
better error messages

Commit:
added new line to end of file

allaptr committed:
Logrus and passing 'name' into client request

allaptr committed:
Add gRPC logging instructions to README

allaptr committed:
Add gRPC logging instructions to README

alex-wzm committed:
Update instructions

alex-wzm committed:
Upgrade tooling

alex-wzm commented on README.md:
```diff
@@ -18,6 +18,10 @@ Currently, the project consists of a Go microservice and a corresponding client
+### Debug gRPC
+Set env variables
+`export GRPC_GO_LOG_VERBOSITY_LEVEL=99`
+`export GRPC_GO_LOG_VERBOSITY_LEVEL=info`
```
Let's create an `.env` file for these. If they should be specific to either the client or server, then put the `.env` in their respective dirs, if not then keep it in root.

allaptr replied:
I think I found a better solution. Will have it configurable on the server programmatically with a flag. Will push shortly.

alex-wzm commented on client-go/cmd/client/main.go:
```diff
@@ -11,6 +12,8 @@ import (
 
 func main() {
+	name := flag.String("name", "Foo-Fighter", "Pass arbitrary string to gRPC call request")
+	flag.Parse()
```
Nice and straight forward input. Please add instructions to `./client-go/README.md`

alex-wzm commented on service-go/cmd/service/service.go:
```diff
@@ -2,9 +2,10 @@ package service
-	"log"
 	"net"
 
+	log "github.com/sirupsen/logrus"
```
I believe logrus needs to be initialized (you would get a runtime error). So instead of renaming "logrus" as "log", instantiate a logger using
```go
log := logrus.New()
```

allaptr replied:
This came straight from logrus README, please check out https://github.com/sirupsen/logrus#example

Creating logger instances is an option, but not necessary. It's recommended if we need `logging to multiple locations from the same application` 
Let's decide which one works best for our case during the interview.

alex-wzm reviewed (changes requested):
Thanks for your contributions @allaptr ! I've left some comments for you to address.

Additionally, I've made some changes in the upstream that may or may not be hard to resolve so let's work on that together during our next interview.

Cheers!

alex-wzm committed:
Update CONTRIBUTING.md

alex-wzm committed:
Update README.md

alex-wzm committed:
Re-track file
* was unintentionally ignored by trying to ignore built binaries

allaptr committed:
PR feedback changes

allaptr committed:
Using .env to set env vars

allaptr committed:
Containerized server

allaptr committed:
README and gitignore

allaptr commented:
I have merged the upstream changes locally, can push them after/during the review

allaptr committed:
Logrus and passing 'name' into client request

allaptr committed:
PR feedback changes

allaptr committed:
Using .env to set env vars

allaptr committed:
Containerized server

allaptr committed:
README and gitignore

allaptr committed:
README and gitignore

allaptr committed:
merged upstream changes

allaptr committed:
merged upstream changes

allaptr committed:
Still merging

allaptr committed:
add jwt validation

allaptr committed:
Fix interceptor opts typo

allaptr committed:
using init to read config file

allaptr committed:
Cleanup repo

allaptr committed:
grpc config added

allaptr committed:
added env

allaptr committed:
Logrus and passing 'name' into client request

allaptr committed:
PR feedback changes

allaptr committed:
Using .env to set env vars

allaptr committed:
Containerized server

allaptr committed:
Still merging

allaptr committed:
Port 8080->50051 and other minor changes

allaptr committed:
Lost and found changes, whew!

allaptr commented:
- I modified loading local.env with godotenv rather than relying on 'source *.env'.
- Added a flag to the service to turn on JWT auth. We can remove it when the client side is implemented.
- Tested with both docker and running service executable directly. 
- Some other minor changes, such as returning errors rather than logging them inside the methods.
I will check for your feedback later, and make fixes if necessary.

allaptr committed:
Removed duplicate file
